cli/mfg.c

```diff
@@ -404,12 +412,87 @@ static int image_select(int argc, char **argv)
 	return ret;
 }
 
+static int secure_state_set(int argc, char **argv)
+{
+	int ret;
+	enum switchtec_boot_phase phase_id;
+	struct switchtec_security_cfg_stat state = {};
+
+	const char *desc = "Set device secure state (BL1 and Main Firmware only)\n\n"
+			   "This command is can only be used when device "
+			   "secure state is UNINITIALIZED_UNSECURED.";
+	static struct {
+		struct switchtec_dev *dev;
+		enum switchtec_secure_state state;
+		int assume_yes_otp;
+	} cfg = {
+		.state = SWITCHTEC_SECURE_STATE_UNKNOWN,
+	};
+	const struct argconfig_options opts[] = {
+		DEVICE_OPTION_NO_PAX,
+		{"state", 't', "state",
+			CFG_CHOICES, &cfg.state,
+			required_argument, "secure state",
+			.choices=secure_state_choices},
+		{"otp", 'o', "", CFG_NONE, &cfg.assume_yes_otp, no_argument,
+		 "assume yes for prompt on OTP programming"},
+		{NULL}
+	};
+
+	argconfig_parse(argc, argv, desc, opts, &cfg, sizeof(cfg));
+
+	if (cfg.state == SWITCHTEC_SECURE_STATE_UNKNOWN) {
+		fprintf(stderr,
+			"Secure state must be set in this command!\n");
+		return -1;
+	}
+
+	ret = switchtec_get_device_info(cfg.dev, &phase_id, NULL, NULL);
+	if (ret < 0) {
```

Comment:
We already have the phase cached to dev in open. Probably need a rebase to have those.

Reply:
Will update once for all, after all commands checked in.